test/SkipCallsTestHook.t.sol

```diff
@@ -0,0 +1,240 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.20;
+
+import {Test} from "forge-std/Test.sol";
+import {Vm} from "forge-std/Vm.sol";
+import {PoolId, PoolIdLibrary} from "../src/types/PoolId.sol";
+import {Hooks} from "../src/libraries/Hooks.sol";
+import {SwapFeeLibrary} from "../src/libraries/SwapFeeLibrary.sol";
+import {IPoolManager} from "../src/interfaces/IPoolManager.sol";
+import {IProtocolFees} from "../src/interfaces/IProtocolFees.sol";
+import {IHooks} from "../src/interfaces/IHooks.sol";
+import {PoolKey} from "../src/types/PoolKey.sol";
+import {PoolManager} from "../src/PoolManager.sol";
+import {PoolSwapTest} from "../src/test/PoolSwapTest.sol";
+import {Deployers} from "./utils/Deployers.sol";
+import {GasSnapshot} from "forge-gas-snapshot/GasSnapshot.sol";
+import {Currency, CurrencyLibrary} from "../src/types/Currency.sol";
+import {MockERC20} from "solmate/test/utils/mocks/MockERC20.sol";
+import {Constants} from "../test/utils/Constants.sol";
+import {SkipCallsTestHook} from "../src/test/SkipCallsTestHook.sol";
+
+contract SkipCallsTest is Test, Deployers, GasSnapshot {
+    using PoolIdLibrary for PoolKey;
+
+    IPoolManager.SwapParams swapParams =
+        IPoolManager.SwapParams({zeroForOne: true, amountSpecified: -100, sqrtPriceLimitX96: SQRT_RATIO_1_2});
+
+    PoolSwapTest.TestSettings testSettings =
+        PoolSwapTest.TestSettings({withdrawTokens: true, settleUsingTransfer: true, currencyAlreadySent: false});
+
+    function deploy(SkipCallsTestHook skipCallsTestHook) public {
+        SkipCallsTestHook impl = new SkipCallsTestHook();
+        vm.etch(address(skipCallsTestHook), address(impl).code);
+        deployFreshManagerAndRouters();
+        skipCallsTestHook.setManager(IPoolManager(manager));
+
+        (currency0, currency1) = deployMintAndApprove2Currencies();
```

Comment:
Why not just put all of this in setUp and then do the init/addLiq as needed in each of the test cases? I think it may be a bit easier to follow what exactly is being called to understand the expected count amount. And I dont think the hook address needs to change each time ie all of this can just happen once? Unless Im missing something

Reply:
so make one hook and turn all flags on? and check that count is total 10 at the end?